datafusion/src/physical_plan/parquet.rs

```diff
@@ -555,21 +625,29 @@ fn build_row_group_predicate(
         row_group_metadata,
         parquet_schema,
     };
-
     let predicate_values = predicate_builder.prune(&pruning_stats);
 
-    let predicate_values = match predicate_values {
-        Ok(values) => values,
+    match predicate_values {
+        Ok(values) => Box::new(move |_, i| {
+            // NB: false means don't scan row group
+            if !values[i] {
```

Comment:
should the counting of filtered row groups happen in the actual filter function? what is the benefit of doing the counting inside the filter function? why not move it outside, just before the filter function is returned?

Reply:
One difference that results from updating the metrics right before use is what happens in a `limit` query -- ala `SELECT * from parquet_table where date < 2012-20-20 limit 10` -- in this case the  query might never even contemplate reading a particular partition due to the limit

However, having written that I think it would make the statistics easier to understand (and more consistent query to query) to report the stats prior to the actual function. I will make that change